templates.rst

```diff
@@ -0,0 +1,1061 @@
+.. index::
+   single: Templating
+
+Creating and Using Templates
+============================
+
+A template is a text file that can generate any text-based format (HTML, XML,
+CSV, etc.). For example, in a web application that displays some news, you can
+create a text file (e.g. ``news.php``) with the HTML structure and PHP logic of
+all news pages. Then, a :doc:`Symfony controller </controllers>` can *render*
+that ``news.php`` template passing the contents of each news to generate the
+resulting HTML contents sent to the user.
+
+Templates were traditionally created with PHP, but in Symfony applications they
+are created with Twig, a flexible, fast, and secure template engine.
+
+.. _twig-language:
+
+Twig Templating Language
+------------------------
+
+`Twig`_ templating language allows you to write concise, readable templates that
+are more friendly to web designers and, in several ways, more powerful than PHP
+templates. Take a look at the following Twig template example. Even if it's the
+first time you see Twig, you probably understand most of it:
+
+.. code-block:: html+twig
+
+    <!DOCTYPE html>
+    <html>
+        <head>
+            <title>Welcome to Symfony!</title>
+        </head>
+        <body>
+            <h1>{{ page_title }}</h1>
+
+            {# this is the main navigation menu #}
+            <ul id="navigation">
+                {% for item in navigation %}
+                    <li><a href="{{ item.href }}">{{ item.caption }}</a></li>
+                {% endfor %}
+            </ul>
+        </body>
+    </html>
```

Comment:
I think we should only show the h1 through the ul (not the whole template body) because it's not realistic to have the entire HTML template inside a single file.

Reply:
I've simplified the example a bit ... but I haven't removed everything, because the purpose of this example is to have at least one occurrence of each of the 3 main Twig constructs: ``{{ ... }}``, ``{% ... %}`` and ``{# ... #}``.